Orange/canvas/application/settings.py

```diff
@@ -359,6 +360,24 @@ def __setupUi(self):
 
         tab.setLayout(form)
 
+        # Error Reporting Tab
+        tab = QWidget()
+        self.addTab(tab, self.tr("Error Reporting"),
+                    toolTip="Settings related to error reporting")
+
+        class MachineIDLineEdit(QLineEdit):
+            def __init__(self):
+                super().__init__()
+                self.setText(get_machine_id())
+
+            def focusOutEvent(self, QFocusEvent):
+                super().focusOutEvent(QFocusEvent)
+                set_machine_id(self.text())
+
+        form = QFormLayout()
+        form.addRow("Machine ID:", MachineIDLineEdit())
+        tab.setLayout(form)
```

Comment:
How about more like the surrounding code:
```py
self.bind(machine_id_edit, 'text', 'debug/machine-id')
```
And then you don't need the helper module _Orange/misc/reporting.py_; you can use:
```py
QSettings().value('debug/machine-id', '', type=str) or str(uuid.getnode())
```